maven-core/src/main/java/org/apache/maven/lifecycle/internal/LifecycleDependencyResolver.java

```diff
@@ -90,7 +90,11 @@ public LifecycleDependencyResolver(
 
     public static List<MavenProject> getProjects(MavenProject project, MavenSession session, boolean aggregator) {
         if (aggregator && project.getCollectedProjects() != null) {
```

Comment:
Here we have check `project.getCollectedProjects() != null` 
but in method `getProjectAndSubModules` we list project `recursively` and we don't check if subproject has `getCollectedProjects() != null` - it is ok? line 102/106

Reply:
I think it can be only on "top level" null (ie. some goal invocation not project being built), and once top level is not null, no child will have it null either... this is just a wild guess.

Reply:
It seems this is really "just implementation details" (that when top level project collectedProjects is not null, then there is no project instance that would have collectedProjects null). So added nullcheck just to be on safe side.